### PR TITLE
Auto load the required gems directly in the providers

### PR DESCRIPTION
**Pull request by tas50**

### CODE REVIEW

**Comment by tas50** [comment]
Resolves #84

File: libraries/lvm.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)
+    end
+
+    chef_gem 'di-ruby-lvm' do
+      action :install
+      version node['lvm']['di-ruby-lvm']['version']
+      compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)
+    end
+
+    require 'lvm'
```
Comment (MarkGibbons) [inline libraries/lvm.rb]: If compile_time isn't supported for the chef_gem resources, say for some version of chef-client 11, won't "require lvm" fail?  Maybe the chef_gem resources could use run_action instead.

File: libraries/lvm.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Chef::Log.debug("Node had di-ruby-lvm-attrib #{node['lvm']['di-ruby-lvm-attrib']['version']} and di-ruby-lvm #{node['lvm']['di-ruby-lvm']['version']} installed. No need to install gems.")
+  rescue LoadError
+    Chef::Log.debug('Did not find lvm gems of the specified versions installed. Installing now')
+
+    chef_gem 'di-ruby-lvm-attrib' do
+      action :install
+      version node['lvm']['di-ruby-lvm-attrib']['version']
+      compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)
+    end
```
Comment (MarkGibbons) [inline libraries/lvm.rb]: Most of my servers are behind firewalls without access to rubygems.org.  Chef_gem looks like it adds a --source=https://rubygems.org by default.  So I would have to continue using a separate recipe to install the gems.  Adding a source option setting would make this change useful for me.

File: libraries/lvm.rb (hunk 1 above)
Comment (tas50) [inline libraries/lvm.rb]: This will increase the requirement for the cookbook for Chef 12

File: libraries/lvm.rb (hunk 2 above)
Comment (tas50) [inline libraries/lvm.rb]: In the scenario where you need a source this change will require you to perform a compile time gem install before using the resource.  Outside of an attribute to specify the source there's no real way around this.

File: libraries/lvm.rb (hunk 1 above)
Comment (CloCkWeRX) [inline libraries/lvm.rb]: If we do an appropriate version bump is that too much of a bad thing?

**Pull request merged**